src/vm/methodtable.cpp

```diff
@@ -2458,6 +2465,67 @@ bool MethodTable::ClassifyEightBytesWithManagedLayout(SystemVStructRegisterPassi
             continue;
         }
 
+        if (fieldClassificationType == SystemVClassificationTypeTypedReference || 
+            CorInfoType2UnixAmd64Classification(GetClass_NoLogging()->GetInternalCorElementType()) == SystemVClassificationTypeTypedReference)
+        {
+            // The TypedReference is a very special type.
```

Comment:
I think that the right way to handle this is to ensure that it is represented appropriately during morph, And then downstream it should just be handled in a standard manner. I don't understand why this additional complexity is necessary.